Refactor(eos_designs): Wildcard dict to list l3_edge (#1784)
aristanetworks/ansible-avd: ansible_collections/arista/avd/roles/eos_designs/doc/l3-edge-v4.0.md

```yaml
l3_edge:
  p2p_links_ip_pools:

>     - pool_name: < p2p_pool_name_1 >

Reviewer: ```suggestion
    - name: < p2p_pool_name_1 >
```

      ip_address: < IPv4_address/Mask >
  p2p_links_profiles:

>     - profile_name: < p2p_profile_name >

Reviewer: ```suggestion
    - name: < p2p_profile_name >
```

Author: fixed

>       variable: < any variable supported under p2p_links can be inherited from a profile >

Reviewer: ```suggestion
      <key>: < any setting supported under p2p_links can be inherited from a profile >
```
